quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/UpdateCommandDistTest.java

```diff
@@ -107,12 +113,109 @@ public void testWrongVersions(KeycloakDistribution distribution) throws IOExcept
         info.put(CachingCompatibilityMetadataProvider.ID, Map.of(
                 "version", "0.0.0.Final",
                 "persistence", "true",
-                "mode", "embedded"
+                "mode", "embedded",
+                "jgroupsVersion", org.jgroups.Version.printVersion()
         ));
         JsonSerialization.mapper.writeValue(jsonFile, info);
 
+        // incompatible jgroups version
         result = distribution.run(UpdateCompatibility.NAME, UpdateCompatibilityCheck.NAME, UpdateCompatibilityCheck.INPUT_OPTION_NAME, jsonFile.getAbsolutePath());
-        result.assertError("[%s] Rolling Update is not available. '%s.version' is incompatible: 0.0.0.Final -> %s.".formatted(CachingCompatibilityMetadataProvider.ID, CachingCompatibilityMetadataProvider.ID, org.infinispan.commons.util.Version.getVersion()));
+        result.assertExitCode(CompatibilityResult.ExitCode.RECREATE.value());
+        result.assertError("[%s] Rolling Update is not available. '%s.version' is incompatible: 0.0.0.Final -> %s.".formatted(CachingCompatibilityMetadataProvider.ID, CachingCompatibilityMetadataProvider.ID, org.infinispan.commons.util.Version.getVersion())); // incompatible infinispan version
+
+        info.put(KeycloakCompatibilityMetadataProvider.ID, Map.of("version", Version.VERSION));
+        info.put(CachingCompatibilityMetadataProvider.ID, Map.of(
+                "version", org.infinispan.commons.util.Version.getVersion(),
+                "persistence", "true",
+                "mode", "embedded",
+                "jgroupsVersion", "0.0.0.Final"
+        ));
+        JsonSerialization.mapper.writeValue(jsonFile, info);
+
+        result = distribution.run(UpdateCompatibility.NAME, UpdateCompatibilityCheck.NAME, UpdateCompatibilityCheck.INPUT_OPTION_NAME, jsonFile.getAbsolutePath());
+        result.assertExitCode(CompatibilityResult.ExitCode.RECREATE.value());
+        result.assertError("[%s] Rolling Update is not available. '%s.jgroupsVersion' is incompatible: 0.0.0.Final -> %s.".formatted(CachingCompatibilityMetadataProvider.ID, CachingCompatibilityMetadataProvider.ID, org.jgroups.Version.printVersion()));
+    }
+
+    @Test
+    public void testCompatibleForV2Feature(KeycloakDistribution distribution) throws IOException {
+        var jsonFile = createTempFile("test-patch-releases-compatible");
+
+        // test the same version
+        testCompatibleV2KeycloakVersion(distribution, jsonFile, Version.VERSION);
+
+        // Test previous micro release
+        ModelVersion modelVersion = new ModelVersion(Version.VERSION);
+        String qualifier = Version.VERSION.split("-", 2)[1];
+        // We are not able to test the following unless we are in a micro version larger than 0
+        //  This is tested in unit test class KeycloakCompatibilityMetadataProvider
+        assumeTrue(modelVersion.getMicro() != 0);
```

Comment:
I wonder if it's worth just skipping these test cases and rely on the unit test? I'm just thinking that it might be confusing if we have different test behaviour at different parts of the lifecycle.

Did you test these cases work by changing the project version locally?

Reply:
I am not aware of any way to change micro version within the distribution tests. I tried to set some environment variable but it doesn't work. The version is hardcoded in `keycloak-common` jar file.

I was thinking we could change the version in main to `999.999.999-SNAPSHOT` (this is what I tested locally and tests passed) but it could be a big change. I just realized we have 999.0.0 also in release branches, this means tests would basically never run :/. I will start some discussion with wider team on this topic. Maybe there is some way to change it.